src/openms/source/CHEMISTRY/ElementDB.cpp

```diff
@@ -640,53 +722,60 @@ namespace OpenMS
+    bool stable_only = half_lifes.empty();
+    for (const auto& m : mass)
     {
-      double atomic_mass = isotope.getMZ();
+      double atomic_mass = m.second;
       unsigned int mass_number = std::round(atomic_mass);
       string iso_name = "(" + std::to_string(mass_number) + ")" + name;
       string iso_symbol = "(" + std::to_string(mass_number) + ")" + symbol;
 
-      // set avg and mono to same value for isotopes (old hack...)
-      double iso_avg_weight = mass.at(mass_number);
-      double iso_mono_weight = iso_avg_weight;
-      IsotopeDistribution iso_isotopes;
-      IsotopeDistribution::ContainerType iso_container;
-      iso_container.push_back(Peak1D(atomic_mass, 1.0));
-      iso_isotopes.set(iso_container);  
-
-      auto iso_element = make_unique<const Element>(iso_name, iso_symbol, an, iso_avg_weight, iso_mono_weight, iso_isotopes);
-      if (auto has_elem = names_.find(iso_name); has_elem != names_.end())
-      { // already exists: overwrite (affects all maps, since they all point to the same thing)
-        overwrite(has_elem->second, iso_element);
-        // do not release 'iso_element' here; it needs to be deleted when it goes out if scope
-      }
-      else
+      double half_life = -1;
+      Isotope::DecayMode dm = Isotope::DecayMode::NONE;
+      if (!stable_only)
       {
-        addIfUniqueOrThrow(names_, iso_name, iso_element);
-        addIfUniqueOrThrow(symbols_, iso_symbol, iso_element);
-        iso_element.release(); // allocation will be cleaned up by ~ElementDB now
+        half_life = half_lifes.at(m.first);
+        dm = decay_modes.at(m.first);
       }
-    } 
+
+      const Isotope* isotope = new Isotope(iso_name, iso_symbol, an, mass_number - an, atomic_mass, abundance.at(m.first), half_life, dm);
+      isotope = addIsotopeToMaps_(iso_name, iso_symbol, isotope);
+      isotopes.push_back(isotope);
+    }
+    return isotopes;
   }
 
-  IsotopeDistribution ElementDB::parseIsotopeDistribution_(const map<unsigned int, double>& abundance, const map<unsigned int, double>& mass)
+  const Isotope* ElementDB::addIsotopeToMaps_(const std::string& iso_name, const std::string& iso_symbol, const Isotope* isotope)
   {
-    IsotopeDistribution::ContainerType dist;
-    
-    for (map<unsigned int, double>::const_iterator it = abundance.begin(); it != abundance.end(); ++it)
-    { 
-      dist.push_back(Peak1D(mass.at(it->first) , abundance.at(it->first)));
+    if (isotopes_.find(iso_symbol) != isotopes_.end())
+    {
+      // TODO: check that lookup is still valid after this
+      const Isotope* const_iso = isotopes_[iso_symbol];
+      Isotope* iso = const_cast<Isotope*>(const_iso);
+      *iso = *isotope; // copy all data from input to the existing element
+      delete isotope;
+      return iso;
+    }
+    else
+    {
+      names_[iso_name] = isotope;
+      symbols_[iso_symbol] = isotope;
+      isotopes_[iso_symbol] = isotope;
+      isotopes_[iso_name] = isotope;
```

Comment:
we should probably do the same things we do here as for the elements to make sure that the lookup still works